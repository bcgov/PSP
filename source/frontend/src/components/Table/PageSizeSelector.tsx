import { IMenuItemProps, Menu } from 'components/menu/Menu';
import React from 'react';
import Form from 'react-bootstrap/Form';
import styled from 'styled-components';

/** align text to middle of size input */
const StyledText = styled.span`
  margin-top: 0.3rem;
`;

interface IProps {
  options: number[];
  value: number;
  onChange: (size: number) => void;
  alignTop: boolean;
}

export const TablePageSizeSelector: React.FC<React.PropsWithChildren<IProps>> = ({
  options,
  value,
  onChange,
  alignTop,
}) => {
  const [selected, setSelected] = React.useState(value);

  const handleValueChange = (newSelection: number) => {
    if (newSelection !== selected) {
      setSelected(newSelection);
      onChange(newSelection);
    }
  };

  const pageSizeOptions: IMenuItemProps[] = options.map(option => ({
    label: option,
    value: option,
    onClick: () => handleValueChange(option),
  }));
  return (
    <Menu options={pageSizeOptions} alignTop={alignTop}>
      <div style={{ display: 'flex' }}>
        <StyledText>Show</StyledText>
        {/*the selector appears to capture the click event which prevents the page change from being registered*/}
        <Form.Control
          size="sm"
          value={`${selected}`}
          type="number"
          style={{ width: 50, marginLeft: 10, marginRight: 10, backgroundColor: 'white' }}
          disabled
        />
        <StyledText>Entries</StyledText>
      </div>
    </Menu>
  );
};
