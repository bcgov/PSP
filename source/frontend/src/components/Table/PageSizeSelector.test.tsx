import React from 'react';
import { create } from 'react-test-renderer';

import { TablePageSizeSelector } from './PageSizeSelector';

const componentRender = () => {
  let component = create(
    <div>
      <TablePageSizeSelector alignTop value={1} options={[1, 2, 3, 4, 5]} onChange={() => {}} />
    </div>,
  );
  return component;
};

describe('Page size selector', () => {
  it('Snapshot matches', () => {
    const component = componentRender();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
