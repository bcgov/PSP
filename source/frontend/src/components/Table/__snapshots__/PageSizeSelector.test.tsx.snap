// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page size selector Snapshot matches 1`] = `
.c0 {
  margin-top: 0.3rem;
}

<div>
  <div
    className="Menu-root"
    onClick={[Function]}
    onFocus={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="Menu-button"
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <span
          className="c0"
        >
          Show
        </span>
        <input
          className="form-control form-control-sm"
          onChange={[Function]}
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
              "width": 50,
            }
          }
          type="number"
          value="1"
        />
        <span
          className="c0"
        >
          Entries
        </span>
      </div>
    </div>
    <div
      className="Menu-items top"
      style={
        Object {
          "width": "10.0rem",
        }
      }
    >
      <div
        className="Menu-options scrollable list-group"
        onKeyDown={[Function]}
      >
        <div
          className="Menu-item list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
          title="menu-item-1"
        >
          1
        </div>
        <div
          className="Menu-item list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
          title="menu-item-2"
        >
          2
        </div>
        <div
          className="Menu-item list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
          title="menu-item-3"
        >
          3
        </div>
        <div
          className="Menu-item list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
          title="menu-item-4"
        >
          4
        </div>
        <div
          className="Menu-item list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
          title="menu-item-5"
        >
          5
        </div>
      </div>
    </div>
  </div>
</div>
`;
